client/js/components/pages/snippets-index.jsx

```diff
@@ -1,35 +1,70 @@
 import React from 'react';
-import AltContainer from 'alt/AltContainer.js';
 import RaisedButton from 'material-ui/lib/raised-button';
 import PageWrapper from '../page-wrapper';
 import SnippetsList from '../snippets/snippets-list';
+import SearchBar from '../search-bar';
 import SnippetFormDialog from '../snippets/snippet-form-dialog';
-import SnippetStore from '../../stores/snippet-store';
 import SnippetActions from '../../actions/snippet-actions';
+import SnippetStore from '../../stores/snippet-store';
+import SnippetSearchStore from '../../stores/snippet-search-store';
 import {langs} from '../../libs/languages';
 
-class SnippetsIndex extends React.Component {
+export default class SnippetsIndex extends React.Component {
   constructor(props) {
     super(props);
+    this.state = this.getPropsFromStores();
+    this._searchSnippets = this._searchSnippets.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onSearch = this.onSearch.bind(this);
   }
 
-  componentDidMount () {
+  componentDidMount() {
+    this.storeListeners = [];
+    this.storeListeners.push(SnippetStore.listen(this.onChange));
+    this.storeListeners.push(SnippetSearchStore.listen(this.onSearch));
     SnippetActions.getAll();
   }
 
+  getPropsFromStores() {
+    return SnippetStore.getState();
+  }
+
+  getPropsFromSearchStore() {
+    return SnippetSearchStore.getState();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(this.getPropsFromStores(nextProps, this.context));
+  }
+
+  componentWillUnmount() {
+    this.storeListeners.forEach(unlisten => unlisten());
```

Comment:
@blimusiek from alt documentation 
`The listen method returns a function that you can use to unsubscribe to store updates.`

```this.storeListeners.push(SnippetStore.listen(this.onChange));
    this.storeListeners.push(SnippetSearchStore.listen(this.onSearch));```